### PR TITLE
network: add betweenness centrality function

### PR DESCRIPTION
**Pull request by kortschak**
The results for examples graphs used from previous tests appear correct although I have not rigorously gone through the calculation by hand. The small cases are clearly correct.

@gonum/developers PTAL

### CODE REVIEW

File: internal/linear.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	if q.Len() == 0 {
+		panic("queue: empty queue")
+	}
+
+	var n graph.Node
+	n, q.data[q.head] = q.data[q.head], nil
+	q.head++
+
+	if q.Len() == 0 {
+		q.head = 0
```
Comment (vladimir-ch) [inline internal/linear.go]: Call q.Reset() instead?

File: network/brandes.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		}
+
+		for _, t := range nodes {
+			sigma[t.ID()] = 0
+			d[t.ID()] = -1
+		}
+		sigma[s.ID()] = 1
+		d[s.ID()] = 0
+
+		queue.Reset()
```
Comment (vladimir-ch) [inline network/brandes.go]: Is this call necessary? At the beginning the queue is empty and at the end of each iteration it is also empty.

File: network/brandes.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	cb := make(map[int]float64, len(nodes))
+	for _, n := range nodes {
+		cb[n.ID()] = 0
+	}
+
+	var (
+		stack internal.NodeStack
+		p     = make(map[int][]graph.Node, len(nodes))
+		sigma = make(map[int]float64, len(nodes))
+		d     = make(map[int]float64, len(nodes))
```
Comment (vladimir-ch) [inline network/brandes.go]: float64 seems unnecessary, int is enough, even better.

**Comment by vladimir-ch** [comment]
If you trust my LGTM on this, then LGTM :-)

**Comment by vladimir-ch** [comment]
Of course after considering my comments above.

File: internal/linear.go (hunk 1 above)
Comment (kortschak) [inline internal/linear.go]: Could, but I'll leave it. Semantically, it's not a reset, but rather just a repositioning that looks like a reset.

**Pull request merged**